user.tf

```diff
@@ -1,14 +1,25 @@
 resource "random_password" "sql_user" {
-  for_each = local.users
+  for_each = merge(
+    local.mysql_users,
+    local.postgres_users,
+  )
 
   length = 48
 }
 
-resource "google_sql_user" "sql_user" {
-  for_each = local.users
+resource "google_sql_user" "mysql_user" {
+  for_each = local.mysql_users
 
   instance = google_sql_database_instance.instance.name
   name     = each.value.name
   host     = each.value.host
   password = random_password.sql_user[each.key].result
 }
+
+resource "google_sql_user" "postgres_user" {
+  for_each = local.postgres_users
+
+  instance = google_sql_database_instance.instance.name
+  name     = each.value.name
+  password = random_password.sql_user[each.key].result
+}
```

Comment:
You don't want to use `google_sql_user` for "most" of the psql users (read: devs, etc). Or are those out of scope for this module?

Reply:
Oh, I hadn't noticed the readonly users are currently not created using `google_sql_user`. I propose we use this approach for users that are allowed all privileges and filter the list on the `readonly` option in the user object. For MySQL this was unsupported to start with (because of issues previously encountered).